File: 19. Remove Nth Node From End of List.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* dummy_head = new ListNode(0);
        dummy_head->next = head;
        
        ListNode* fp, *lp;
        fp = lp = dummy_head;
        
        int count = 0;
        
        while (lp != NULL){
            lp = lp->next;
            
            if (count == n + 1)
                fp = fp->next;            
            else
                count++;            
        }
        
        ListNode* tmp = fp->next;
                
        fp->next = tmp->next;
        
        free(tmp);                
        
        head = dummy_head->next;
        
        return head;
    }
};

File: Python/19. Remove Nth Node From End of List.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        dummy = ListNode(0)
        dummy.next = head        
        slow = fast = dummy
        for i in range(n + 1):
            fast = fast.next
        while fast != None:
            slow = slow.next
            fast = fast.next
        slow.next = slow.next.next
        return dummy.next